lib/datadog/tracing/contrib/pg/instrumentation.rb

```diff
@@ -0,0 +1,75 @@
+# typed: false
+
+require 'datadog/tracing'
+require 'datadog/tracing/metadata/ext'
+require 'datadog/tracing/contrib/analytics'
+require 'datadog/tracing/contrib/pg/ext'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Pg
+        # PG::Connection patch module
+        module Instrumentation
+          def self.included(base)
+            base.prepend(InstanceMethods)
+          end
+
+          # PG::Connection patch methods
+          module InstanceMethods
+            def exec(sql)
+              service = Datadog.configuration_for(self, :service_name) || datadog_configuration[:service_name]
+
+              Tracing.trace(Ext::SPAN_EXEC, service: service) do |span|
+                span.resource = sql
+                span.type = Tracing::Metadata::Ext::SQL::TYPE
+                span.service = Ext::DEFAULT_PEER_SERVICE_NAME
```

Comment:
Minor change: let's try to move as many values to the trace method call arguments (e.g. type be passed a a keyword argument to `#trace` for example).

This reduces the amount of times the span is mutated, and can allow for better introspection of the Span at initialization time if any parties are interested in inspecting it (e.g. sampling could do that in the future).

There's a minor performance improvement as well, as we save a method call (e.g. `span.type = Tracing::Metadata::Ext::SQL::TYPE`).